src/lib/steps.ts

```diff
@@ -40,7 +40,7 @@ interface AggFunctionStep {
   /** name of the agg function step, (typically the name of the output column) */
   name: string;
   /** the aggregation operation (e.g. `sum` or `count`) */
-  aggfunction: string;
+  aggfunction?: 'sum' | 'avg' | 'count' | 'min' | 'max';
```

Comment:
I don't see why `aggfunction` should be optional and as far as I can see, the `if / else` branch in the `transformAggregate` function assume it's not empty.

Reply:
Yes you’re right it should not be optional !! I will fix this thank you